execution/src/controller.rs

```diff
@@ -0,0 +1,99 @@
+use crate::config::{ExecutionConfig, CHANNEL_SIZE};
+use crate::error::ExecutionError;
+use crate::worker::{
+    ExecutionCommand, ExecutionEvent, ExecutionManagementCommand, ExecutionWorker,
+};
+use models::{Block, BlockHashMap};
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+
+/// A sender of execution commands.
+#[derive(Clone)]
+pub struct ExecutionCommandSender(pub mpsc::Sender<ExecutionCommand>);
+
+/// A receiver of execution events.
+pub struct ExecutionEventReceiver(pub mpsc::UnboundedReceiver<ExecutionEvent>);
+
+/// A sender of execution management commands.
+pub struct ExecutionManager {
+    join_handle: JoinHandle<Result<(), ExecutionError>>,
+    manager_tx: mpsc::Sender<ExecutionManagementCommand>,
+}
+
+impl ExecutionManager {
+    pub async fn stop(self) -> Result<(), ExecutionError> {
+        drop(self.manager_tx);
+
+        if self.join_handle.await.is_err() {
+            return Err(ExecutionError::JoinError);
+        };
+
+        Ok(())
+    }
+}
+
+/// Creates a new execution controller.
+///
+/// # Arguments
+/// * thread_count: thread count architecture parameter
+/// * cfg: execution configuration
+///
+/// TODO: add a consensus command sender,
+/// to be able to send the `TransferToConsensus` message.
+pub async fn start_controller(
+    cfg: ExecutionConfig,
+    thread_count: u8,
```

Comment:
Why thread count is not in the config ?

Reply:
it is in consensusconfig, we don't want to duplicate it and we don't need the rest of consensusconfig, so we passed it directly

Reply:
Maybe there is something to dig up that's related to #1839 @yvan-sraka